code/modules/mob/living/simple_animal/bot/secbot.dm

```diff
@@ -193,8 +193,6 @@
 
 /mob/living/simple_animal/bot/secbot/attackby(obj/item/W, mob/user, params)
 	..()
-	if(istype(W, /obj/item/weldingtool) && user.a_intent != INTENT_HARM) // Any intent but harm will heal, so we shouldn't get angry.
```

Comment:
Are you also removing non damaging / stamina attack reaction, and not just the welder?

Reply:
This check is two lines below

Reply:
Ah, excuse me then, sorry